src/auth/handlers.rs

```diff
@@ -73,30 +73,40 @@ pub fn signup_post(conn: ObservDbConn, mut cookies: Cookies, form: Form<SignUpFo
 
     use crate::schema::users::dsl::*;
 
-    insert_into(users)
-        .values(&newuser)
-        .execute(&*conn)
-        .expect("Failed to add user to database");
-
-    let user: User = users
-        .filter(&email.eq(newuser.email))
-        .first(&*conn)
-        .expect("Failed to get user from database");
-
+    // Check if user's email is already signed up
+    if let Some(user) = users
```

Comment:
Ah, found something when I went to test this (should have done that first, my bad).

This doesn't need to be an `if let`. You use an `if let` when you want to use the value, in this case `user`. But since we just want to check for existence it would be better to do:

```rust
if users
    .filter(...)
    ...
    .expect(...)
    .is_some()
```

Reply:
I see, just what I was looking for! Will implement this right away.